src/zmalloc.c

```diff
@@ -388,12 +391,15 @@ void zfree_usable(void *ptr, size_t *usable) {
 
     if (ptr == NULL) return;
 #ifdef HAVE_MALLOC_SIZE
-    update_zmalloc_stat_free(*usable = zmalloc_size(ptr));
+    size_t size = zmalloc_size(ptr);
+    update_zmalloc_stat_free(size);
+    if (usable) *usable = size;
```

Comment:
if there's no bug in the old code, i'd rather drop these refactoring

Reply:
ohh right, we agree to drop this, i forgot to drop the code in zfree